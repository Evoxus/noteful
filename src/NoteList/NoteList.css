.NoteList ul{
  list-style-type: none;
  padding-left: 0;
  margin-top: 0;
}
.addNoteButton {
  text-align: center;
  border: 1px solid green;
  width: 100px;
  height: 40px;
  margin: 5px;
  padding: 10px 0;
  cursor: pointer;
}
.addNoteButton:hover{
  background-color: green;
}
.NoteList >ul>li>a {
  width: 100%;
}
.add-button{
  text-decoration: none;
  list-style-type: none;
  color: lightgreen;
  height: 100%;
  width: 100%;
}
