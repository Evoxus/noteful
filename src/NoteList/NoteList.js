import React from 'react';
import Note from '../Note/Note';
import './NoteList.css';
import NoteContext from '../NoteContext';


export default function NoteList(props) {
  return (
    <NoteContext.Consumer>
      {(value) => {
        const notes = value.notes.map(note =>
          <li key={note.id}>
            <Note modified={note.modified}
              name={note.name} id={note.id}
            />
          </li>
        );
        return (
          <section className='NoteList'>
            <ul>
              {notes}
            </ul>
          </section>
        )
      }}
    </NoteContext.Consumer>

  )
}

NoteList.defaultProps = {
  notes: []
}