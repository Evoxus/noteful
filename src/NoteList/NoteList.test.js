import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import NoteList from './NoteList';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter> <NoteList /> </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('snapshot test', () => {
    const wrapper = shallow(<NoteList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})