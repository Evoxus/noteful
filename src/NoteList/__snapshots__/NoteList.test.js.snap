// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<section
  className="NoteList"
>
  <ul>
    <li
      className="addNoteButton"
      key="addNote"
    >
      <Link
        className="add-button"
        to="/AddNote"
      >
        Add Note
      </Link>
    </li>
  </ul>
</section>
`;
